src/axom/sidre/spio/IOManager.cpp

```diff
@@ -84,7 +84,15 @@ IOManager::IOManager(MPI_Comm comm,
 {
   MPI_Comm_size(comm, &m_comm_size);
   MPI_Comm_rank(comm, &m_my_rank);
-#ifndef AXOM_USE_SCR
+#ifdef AXOM_USE_SCR
+  int world_size;
+  MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+  if (m_use_scr && m_comm_size != world_size)
```

Comment:
The text of this warning isn't correct.  What is the behavior we should warn about if m_comm_size does not equal the size of MPI_COMM_WORLD?

Reply:
I am not sure I understand this restriction. Can we not use SCR on a sub-communicator?

Reply:
Actually, I think we can now probably drop this check and its message.  I think I added this before I moved the Start_checkpoint call.  Since I moved the Start_checkpoint call up to the application, it should now work even when the user creates IOManager with a communicator that is smaller than comm world.

Reply:
Most SCR functions are an implied collective across all procs in MPI_COMM_WORLD.  We dup comm_world during SCR_Init.  All procs in the job have to call start/complete, but it's acceptable for a subset of procs to do the actual I/O.  The ranks that read/write data must call SCR_Route_file for each of their files.  I changed the examples so that it's the responsibility of the application to make the collective calls, and then SPIO only has to call Route_file, which is local to the calling process (not a collective).